src/sentry/rules/processing/delayed_processing.py

```diff
@@ -240,10 +238,12 @@ def get_group_to_groupevent(
 @redis_buffer_registry.add_handler(BufferHookEvent.FLUSH)
 def process_delayed_alert_conditions(buffer: RedisBuffer) -> None:
     with metrics.timer("delayed_processing.process_all_conditions.duration"):
-        project_ids = buffer.get_set(PROJECT_ID_BUFFER_LIST_KEY)
-        for project_id in project_ids:
-            with metrics.timer("delayed_processing.process_project.duration"):
-                apply_delayed.delay(project_id=project_id)
+        fetch_time = datetime.now(tz=timezone.utc)
+        project_ids = buffer.get_sorted_set(PROJECT_ID_BUFFER_LIST_KEY)
```

Comment:
`get_sorted_set` needs to incorporate the same range as `delete_key`. So we know that we're only fetching data between `0` and `fetch_time`.

Maybe a bit pedantic, but it helps us avoid a situation where we get the timestamp, then an extra project id is added to the set before we fetch, which we would end up not deleting.

Reply:
Updated but the behavior is kinda strange, `zrange` only accepts integers for `start` and `end` versus how `zremrangebyscore` (the Redis docs say `zrange` replaces `zrangebyscore`) accepts floats (aka what timestamps are) for `min` and `max`.

Reply:
Ahh, we need to use https://redis.io/docs/latest/commands/zrangebyscore/

Reply:
Or actually, https://redis.io/docs/latest/commands/zrange/ and use the `byscore` option if that's available for us

Reply:
Nm, I just saw you already did this, ignore me